### PR TITLE
Split

### PR DESCRIPTION
**Pull request by sunjincheng121**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
1. API Expressions 加Expression后缀
2. 枚举变量用 API expression 里面的
3. 原先的 expression 包名不变
4. API Expression 包路径是否要改成 `api.expression` ?
5. visitor 模式里面的 case FunctionDefinitions.TRIM 改成 case TRIM
6. remove DistinctAgg

```
public interface ExpressionVisitor<R> {

	R visitCall(CallExpress call);
	
	R visitTypeLiteral(TypeLiteralExpression typeLiteral);

	R visitSymbol(SymbolExpression symbolExpression);

	R visitValueLiteral(ValueLiteralExpression literal);

	R visit(Expression other);

	R visitFieldReference(FieldReferenceExperssion fieldReference);

}
```

File: flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/ExpressionVisitor.java [hunk 1]
```diff
@@ -27,18 +27,18 @@
...
 
-	R visitTypeLiteral(TypeLiteral typeLiteral);
+	R visitTypeLiteral(TypeLiteralExpression typeLiteralExpression);
 
 	R visitSymbolExpression(SymbolExpression symbolExpression);
 
-	R visitLiteral(Literal literal);
+	R visitLiteral(ValueLiteralExpression valueLiteralExpression);
 
 	R visitOther(Expression other);
```
Comment (sunjincheng121) [inline flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/ExpressionVisitor.java]: only using visit

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 2]
```diff
@@ -1001,7 +1002,8 @@ abstract class TableEnvironment(val config: TableConfig) {
     * used if the input type has a defined field order (tuple, case class, Row) and no of fields
     * references a field of the input type.
     */
-  protected def isReferenceByPosition(ct: CompositeType[_], fields: Array[Expression]): Boolean = {
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: revert

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 3]
```diff
@@ -1038,9 +1040,10 @@ abstract class TableEnvironment(val config: TableConfig) {
 
   /**
     * Returns field names and field positions for a given [[TypeInformation]] and [[Array]] of
-    * [[Expression]]. It does not handle time attributes but considers them in indices.
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: PlannerExpression继承Expression之后，这个感觉可以revert了？

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala [hunk 4]
```diff
@@ -27,6 +27,8 @@ import org.apache.flink.streaming.api.datastream.DataStream
 import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
 import _root_.java.lang.{Boolean => JBool}
 
+import org.apache.flink.table.expressions.Expression
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala]: import order，应该放到那堆 org.apache.flink.table import里

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/BatchTableEnvironment.scala [hunk 5]
```diff
@@ -80,7 +80,6 @@ class BatchTableEnvironment @deprecated(
     * @return The converted [[Table]].
     */
   def fromDataSet[T](dataSet: DataSet[T], fields: Expression*): Table = {
-
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/BatchTableEnvironment.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataSetConversions.scala [hunk 6]
```diff
@@ -21,6 +21,7 @@ import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.api.scala._
 import org.apache.flink.table.api.Table
 import org.apache.flink.table.expressions.Expression
+import org.apache.flink.table.expressions.PlannerExpression
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataSetConversions.scala]: unused import

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataStreamConversions.scala [hunk 7]
```diff
@@ -19,8 +19,9 @@ package org.apache.flink.table.api.scala
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.table.api.Table
-import org.apache.flink.table.expressions.Expression
+import org.apache.flink.table.expressions.PlannerExpression
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataStreamConversions.scala]: unused, 这个文件的改动可以还原了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/StreamTableEnvironment.scala [hunk 8]
```diff
@@ -20,10 +20,10 @@ package org.apache.flink.table.api.scala
 import org.apache.flink.api.scala._
 import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.table.api.{StreamQueryConfig, Table, TableConfig, TableEnvironment}
-import org.apache.flink.table.expressions.Expression
 import org.apache.flink.table.functions.{AggregateFunction, TableFunction}
 import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
 import org.apache.flink.streaming.api.scala.asScalaStream
+import org.apache.flink.table.expressions.Expression
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/StreamTableEnvironment.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/mathExpressions.scala [hunk 9]
```diff
@@ -488,7 +488,7 @@ case class Hex(child: PlannerExpression) extends UnaryPlannerExpression {
   }
 }
 
-case class UUID() extends LeafPlannerExpression {
+case class PlannerUUID() extends LeafPlannerExpression {
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/mathExpressions.scala]: ?这个为什么要改

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/mathExpressions.scala [hunk 10]
```diff
@@ -387,7 +387,7 @@ case class Pi() extends LeafPlannerExpression {
   }
 }
 
-case class E() extends LeafPlannerExpression {
+case class PlannerE() extends LeafPlannerExpression {
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/mathExpressions.scala]: ?这个为什么要改

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/functions/DistinctAggregateFunction.scala [hunk 11]
```diff
@@ -18,7 +18,7 @@
 package org.apache.flink.table.functions
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
-import org.apache.flink.table.expressions.{AggregateFunctionDefinition, DistinctAgg, Expression, ExpressionUtils}
+import org.apache.flink.table.expressions._
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/functions/DistinctAggregateFunction.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/PlannerExpression.scala [hunk 12]
```diff
@@ -17,14 +17,18 @@
  */
 package org.apache.flink.table.expressions
 
+import java.util
+
 import org.apache.calcite.rex.RexNode
 import org.apache.calcite.tools.RelBuilder
-
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/PlannerExpression.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/PlannerExpression.scala [hunk 13]
```diff
@@ -17,14 +17,18 @@
...
 import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.table.expressions.{Expression, ExpressionVisitor}
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/PlannerExpression.scala]: unused import，同一个package下，不需要import了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/fieldExpression.scala [hunk 14]
```diff
@@ -153,23 +155,27 @@ case class TableReference(name: String, table: Table) extends LeafExpression wit
...
-  extends UnaryExpression
+abstract class TimeAttribute(val expression: PlannerExpression)
+  extends UnaryPlannerExpression
   with WindowProperty {
 
-  override private[flink] def child: Expression = expression
+  override private[flink] def child: PlannerExpression = expression
 }
 
-case class RowtimeAttribute(expr: Expression) extends TimeAttribute(expr) {
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/fieldExpression.scala]: case class RowtimeAttribute里的change都是unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/logic.scala [hunk 15]
```diff
@@ -20,11 +20,10 @@ package org.apache.flink.table.expressions
 import org.apache.calcite.rex.RexNode
 import org.apache.calcite.sql.fun.SqlStdOperatorTable
 import org.apache.calcite.tools.RelBuilder
-
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/logic.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/package.scala [hunk 16]
```diff
@@ -15,7 +15,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.flink.table
+package org.apache.flink.table.plan
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/package.scala]: plan删了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala [hunk 17]
```diff
@@ -23,19 +23,21 @@ import org.apache.calcite.rex._
 import org.apache.calcite.sql.fun.SqlStdOperatorTable
 import org.apache.calcite.tools.RelBuilder
 import org.apache.flink.api.common.typeinfo.BasicTypeInfo._
-import org.apache.flink.api.common.typeinfo.{SqlTimeTypeInfo, TypeInformation}
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala [hunk 18]
```diff
@@ -23,19 +23,21 @@ import org.apache.calcite.rex._
...
+import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.table.calcite.FlinkRelBuilder
-import org.apache.flink.table.expressions.TimeIntervalUnit.TimeIntervalUnit
 import org.apache.flink.table.functions.sql.ScalarSqlFunctions
 import org.apache.flink.table.typeutils.TypeCheckUtils.isTimeInterval
 import org.apache.flink.table.typeutils.{TimeIntervalTypeInfo, TypeCheckUtils}
 import org.apache.flink.table.validate.{ValidationFailure, ValidationResult, ValidationSuccess}
+import org.apache.flink.api.common.typeinfo.SqlTimeTypeInfo._
+import org.apache.flink.table.expressions.TimeIntervalUnit._
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala]: import order

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/RexProgramExtractor.scala [hunk 19]
```diff
@@ -26,11 +26,11 @@ import org.apache.calcite.util.{DateString, TimeString, TimestampString}
 import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, SqlTimeTypeInfo}
 import org.apache.flink.table.api.TableException
 import org.apache.flink.table.calcite.FlinkTypeFactory
-import org.apache.flink.table.expressions.{And, Expression, Literal, Or, ResolvedFieldReference}
 import org.apache.flink.table.validate.FunctionCatalog
 import org.apache.flink.util.Preconditions
 import java.sql.{Date, Time, Timestamp}
 
+import org.apache.flink.table.expressions._
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/util/RexProgramExtractor.scala]: import order

File: flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/AggregateCallExpression.java [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.util.List;
+
+/**
+ * General expression for unresolved Aggregate function calls. The function can be a built-in
+ * function or a user-defined function.
+ */
+@PublicEvolving
+public class AggregateCallExpression extends CallExpression {
```
Comment (dianfu) [inline flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/AggregateCallExpression.java]: final

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala [hunk 21]
```diff
@@ -107,12 +107,12 @@ abstract class TemporalCeilFloor(
 
     (unit.get, temporal.resultType) match {
       case (TimeIntervalUnit.YEAR | TimeIntervalUnit.MONTH,
-          SqlTimeTypeInfo.DATE | SqlTimeTypeInfo.TIMESTAMP) =>
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala [hunk 22]
```diff
@@ -107,12 +107,12 @@ abstract class TemporalCeilFloor(
...
+      SqlTimeTypeInfo.DATE | SqlTimeTypeInfo.TIMESTAMP) =>
         ValidationSuccess
       case (TimeIntervalUnit.DAY, SqlTimeTypeInfo.TIMESTAMP) =>
         ValidationSuccess
       case (TimeIntervalUnit.HOUR | TimeIntervalUnit.MINUTE | TimeIntervalUnit.SECOND,
-          SqlTimeTypeInfo.TIME | SqlTimeTypeInfo.TIMESTAMP) =>
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/time.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/functions/DistinctAggregateFunction.scala [hunk 23]
```diff
@@ -18,12 +18,12 @@
 package org.apache.flink.table.functions
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
-import org.apache.flink.table.expressions.{AggFunctionCall, DistinctAgg, Expression}
+import org.apache.flink.table.expressions.{Expression, ExpressionUtils, FunctionDefinitions, AggregateFunctionDefinition}
 import org.apache.flink.table.functions.utils.UserDefinedFunctionUtils.{getAccumulatorTypeOfAggregateFunction, getResultTypeOfAggregateFunction}
 
 /**
   * Defines an implicit conversion method (distinct) that converts [[AggregateFunction]]s into
-  * [[DistinctAgg]] Expressions.
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/functions/DistinctAggregateFunction.scala]: [[AggregateCallExpression]]

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala [hunk 24]
```diff
@@ -69,12 +69,16 @@ abstract class WindowPropertiesBaseRule(rulePredicate: RelOptRuleOperand, ruleNa
     val timeProperties = windowType match {
       case 'streamRowtime =>
         Seq(
-          NamedWindowProperty(propertyName(w, "rowtime"), RowtimeAttribute(w.aliasAttribute)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala]: 这里面的改动可以revert
